src/main/java/proto/objects.proto

```diff
@@ -0,0 +1,30 @@
+/*
+Instead of encoding all data in one proto, we will instead preface the proto
+with a header field containing the number of total bytes in the proto. This will
+be a configurable value that will be computed where the biggest possible Data
+division below the maximum threshold is used. This will allow for easy chunking
+when reading in messages etc.
+*/
+syntax = "proto3";
+
+package formats;
+option java_package = "proto";
+option java_outer_classname = "DataFormats";
+
+message SerializedData { repeated Data data = 1; }
```

Comment:
Do you need this proto only containts repeated Pose data?
For sstable, each record is a <timestamp, pose>, you don't need this proto of the repeated pose.

Reply:
The original example I was going off uses a format similar to this, I think you are correct that this is unnecessary though I'm curious what the generated json will look like

Reply:
For json, it is another story, not related to your proto definition. 
I think the json will be like this:
pose_data = [
{"timestamp": 123456, "pose":[1, 2, 3, 10, 0, 0]},
{"timestamp": 123457, "pose":[1, 2, 4, 10, 0, 0]},
{"timestamp": 123458, "pose":[1, 2, 5, 10, 0, 0]},
];

Reply:
The build in json converter is based on the proto definition, so assuming we don't write the converter from scratch it would depend on the proto definition.

Reply:
You can use json converter convert one pose data, and added it in your json array.
You'd better not add all data in one proto message, it will be very big, proto message is designed for small data, not you whole run's data.